Source/missiles.cpp

```diff
@@ -4894,17 +4823,14 @@ void MI_Rportal(int i)
 
 void ProcessMissiles()
 {
-	int i, mi;
-
-	for (i = 0; i < nummissiles; i++) {
+	for (int i = 0; i < nummissiles; i++) {
 		dFlags[missile[missileactive[i]].position.tile.x][missile[missileactive[i]].position.tile.y] &= ~BFLAG_MISSILE;
 		dMissile[missile[missileactive[i]].position.tile.x][missile[missileactive[i]].position.tile.y] = 0;
 		if (missile[missileactive[i]].position.tile.x < 0 || missile[missileactive[i]].position.tile.x >= MAXDUNX - 1 || missile[missileactive[i]].position.tile.y < 0 || missile[missileactive[i]].position.tile.y >= MAXDUNY - 1)
 			missile[missileactive[i]]._miDelFlag = true;
 	}
 
-	i = 0;
-	while (i < nummissiles) {
+	for (int i = 0; i < nummissiles;) {
 		if (missile[missileactive[i]]._miDelFlag) {
 			DeleteMissile(missileactive[i], i);
 			i = 0;

```

Comment:
```suggestion
```
This assignment is unnecessary, the DeleteMissile function only modifies `missileactive[i]`